qa-tests-backend/src/test/groovy/BaseSpecification.groovy

```diff
@@ -28,14 +30,15 @@ import util.Env
 import util.Helpers
 import util.OnFailure
 
+import org.junit.AssumptionViolatedException
 import org.junit.Rule
 import org.junit.rules.TestName
 import org.junit.rules.Timeout
 import spock.lang.Retry
 import spock.lang.Shared
 import spock.lang.Specification
 
-@Retry(condition = { Helpers.determineRetry(failure) })
+@Retry(condition = { Object failure -> determineRetry(failure as Throwable) }, mode = SETUP_FEATURE_CLEANUP)
```

Comment:
Does this mean that any common content of `cleanup:` sections within the test cases can now be moved to the `cleanup()` method?

Reply:
Yep, I think that `cleanup:` block should be used only for specific test cleanup all common things should go in `cleanup()`